src/AutoRest.CSharp/Mgmt/Decorator/SingletonDetection.cs

```diff
@@ -14,7 +14,7 @@ namespace AutoRest.CSharp.Mgmt.Decorator
 {
     internal static class SingletonDetection
     {
-        private static string[] SingletonKeywords = { "default", "latest" };
+        private static string[] SingletonKeywords = { "default", "latest", "current" };
```

Comment:
Do we have tests for this?

Reply:
Added a test case for `current` in a new singleton pattern where you do not hardcode the resource name in the API path, but with a single-value enum name parameter.